engine/src/test/java/org/hibernate/search/test/dsl/SortDSLTest.java

```diff
@@ -0,0 +1,809 @@
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "uniqueNumericField" )
+						.desc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 3, 0, 1, 2 )
+		);
+	}
+
+	@Test
+	public void singleField_missingValue_use() throws Exception {
+		Query query = builder().all().createQuery();
+
+		Sort sort = builder().sort()
+				.byField( "uniqueNumericField" )
+						.onMissingValue().use( 1.5d )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 1, 2, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "uniqueNumericField" )
+						.asc()
+						.onMissingValue().use( 1.5d )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 1, 2, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "uniqueNumericField" )
+						.desc()
+						.onMissingValue().use( 1.5d )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 3, 0, 2, 1 )
+		);
+	}
+
+	@Test
+	public void singleField_stringFieldBridge() throws Exception {
+		Query query = builder().all().createQuery();
+
+		// Missing value is not provided; the missing values should be considered as 0
+
+		Sort sort = builder().sort()
+				.byField( "fieldBridgedStringField" )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "fieldBridgedStringField" )
+						.asc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "fieldBridgedStringField" )
+						.desc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 3, 0, 1, 2 )
+		);
+	}
+
+	@Test(expected = SearchException.class)
+	public void singleField_stringFieldBridge_missingValue_use() throws Exception {
+		builder().sort()
+				.byField( "fieldBridgedStringField" )
+						.onMissingValue().use( "1.5" )
+				.createSort();
+	}
+
+	@Test
+	public void singleField_numericFieldBridge() throws Exception {
+		Query query = builder().all().createQuery();
+
+		// Missing value is not provided; the missing values should be considered as 0
+
+		Sort sort = builder().sort()
+				.byField( "fieldBridgedNumericField", SortField.Type.DOUBLE )
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "fieldBridgedNumericField", SortField.Type.DOUBLE )
+						.asc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 2, 1, 0, 3 )
+		);
+
+		sort = builder().sort()
+				.byField( "fieldBridgedNumericField", SortField.Type.DOUBLE )
+						.desc()
+				.createSort();
+		assertThat(
+				query( query, sort ),
+				returnsIDsInOrder( 3, 0, 1, 2 )
+		);
+	}
+
+	@Test
+	public void singleField_numericFieldBridge_missingValue_use() throws Exception {
```

Comment:
It's just for tests. If we want to express the purpose of tests in their name, we have to use exceptionally long names, and underscores help making this readable, or at least less unreadable.

You can find other places where this syntax is used:

 * In QueryDSL (which is arguably a very, very clean piece of software, and happens to only use underscores when naming test methods) : https://github.com/querydsl/querydsl/blob/master/querydsl-jpa/src/test/java/com/querydsl/jpa/CollectionTest.java
 * In Hibernate Validator: https://github.com/hibernate/hibernate-validator/blob/master/engine-jdk8-tests/src/test/java/org/hibernate/validator/test/internal/engine/typeannotationconstraint/TypeAnnotationConstraintTest.java , https://github.com/hibernate/hibernate-validator/blob/master/engine/src/test/java/org/hibernate/validator/test/internal/constraintvalidators/hv/URLValidatorTest.java
 * In this project, though it seems a bit exotic: https://github.com/hibernate/hibernate-search/blob/master/orm/src/test/java/org/hibernate/search/test/spatial/SpatialSearchSortByDistanceAndPagingTest.java

Then again, on that kind of code (tests), I'd value clarity over consistency. But let me know if I must rename these method anyway.